ada_2_fhir-r4/zibs2024/MAPPING_GUIDELINES.md

```diff
@@ -0,0 +1,78 @@
+
+### Specific guidelines
+#### Parameters
+Each profile template SHOULD have an optional parameter called `$in` that contains the ADA object being transformed, defaulting to the current context.
+
+Each profile template SHOULD have an optional parameter called `$subject` containing the ADA object of the patient to which the zib instance applies. Of course, this doesn't apply when the instance is not about a patient.
+
+Additional parameters may be used as desired, as long as they are documented properly.
+
+#### Profile name variable
+For each FHIR profile add a variable of the form `profileName[English zib name]` to the respective template. This variable should be used whenever this profile needs to be referenced, instead of stating the profile name directly. This ensures a smaller risk on typos in the profile name and makes it easier to change a profile name in all relevant templates.
+
+Example: `<xsl:variable name="profileNamePatient">nl-core-Patient</xsl:variable>`
+
+#### Logical id
+In each template that is used to map an ADA instance to a FHIR resource instance, a logical id should be added to the resulting FHIR instance. This is done by invoking the template `insertLogicalId`. Note that the parameter `profile` may be used whenever there is no one-on-one relation between the ADA element and FHIR resource.
+
+Example:
+
+                <xsl:call-template name="insertLogicalId">
+                    <xsl:with-param name="profile" select="$profileNamePatient"/>
+                </xsl:call-template>
+
+                
+#### `meta.profile` element
+
+In each profile template, the `.meta.profile` element should be added to the resulting FHIR instance. This is done by:
+
+* the `nf:get-full-profilename-from-adaelement(.)` function whenever there is an one-on-one relation between the ada element and FHIR resource (see the variable `ada2resourceType` in `2_fhir_fhir_include.xsl`);
+* `concat($urlBaseNictizProfile, $profileName[English zib name])` in all other cases.
+
+
+### Zibs resulting in multiple FHIR resources
+Sometimes, the transformation of a single ADA instance can result in multiple FHIR resources. For example, zib HeartRate has a focal Observation profile that represents the zib itself, and two associated Observation profiles for the concepts HeartbeatRegularity and InterpretationHeartRate (which are referenced from the focal Observation).
+
+As described above, a profile template is created for each resource that needs to be created, with name and mode set to the profile id and a match on the corresponding ADA element. The template for the focal profile normally acts as the entry point for the transformation. So in addition to producing the focal resource, it is also responsible for kicking off the transformations to the associated resources. The result of the transformation is thus a _list_ of resources.
```

Comment:
I've changed _list_ to _set_ in the last sentence to avoid possible confusion with the FHIR List resource (which might be a bit far-fetched).

Reply:
It's probably correcter in XSLT terms as well ...